### PR TITLE
[4.0] Backport skip unready nodes deployment

### PR DESCRIPTION
**Pull request by Itxaka**
If we set the experimental skip_unready_nodes config to true and provide
a list of roles, we try to skip nodes that are not ready from the
deployment of the barclamp so we dont need all nodes to be ready to
deploy a barclamp.

This is useful when dealing with a high number of nodes (i.e.
nova-compute) and we dont need all those other nodes to be ready to add
a new node to a role

Background:

When changing the DNS forwarder in the DNS barclamp all nodes of the cloud must be ready and accessible to apply the change, although the change does not affect all nodes.

There must be a way to only target affected or a selected sub-set of nodes:

just ignore not ready/not accessible nodes (they will be automatically updated on their next local chef-client run)

Backport of: https://github.com/crowbar/crowbar-core/pull/1341

### CODE REVIEW

**Comment by Itxaka** [comment]
[Do not require all nodes to be ready for applying proposals](https://trello.com/c/14rO1Ofw/90-do-not-require-all-nodes-to-be-ready-for-applying-proposals)

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -1014,19 +1034,21 @@ def apply_role(role, inst, in_queue, bootstrap = false)
...
-    # when pacemaker barclamp is deactivated.  elements_expanded gets updated
-    # by pacemaker barclamp.
-    old_elements = {}
-    old_deployment = old_role.override_attributes[@bc_name] unless old_role.nil?
-    unless old_deployment.nil?
-      old_elements = old_deployment["elements_expanded"]
-      if old_elements.nil?
-        old_elements = old_deployment["elements"]
+    # make sure the role is saved
+    role.save
```
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: These two lines are wrong; I think that's due to a broken rebase.

Comment (Itxaka) [reply crowbar_framework/app/models/service_object.rb]: Something weird happened here as it showed like you never merged the patch that affected this, even with a rebase. Im guessing that it was a github delay or something like that. Fixed it!

**Comment by vuntz** [comment]
> (cherry picked from commit 493c26b)

This is not the right commit id anymore (in the last commit).

**Comment by Itxaka** [comment]
> This is not the right commit id anymore (in the last commit).
indeed, fixed!

**Pull request merged**